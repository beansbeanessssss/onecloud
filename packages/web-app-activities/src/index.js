import translations from '../l10n/translations'
import Activities from './views/Activities.vue'
// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Activities'),
  id: 'activities',
  icon: 'activities',
  isFileEditor: false
}

const routes = [
  {
    path: '/',
    name: 'activities-all',
    component: Activities,
    meta: {
      title: $gettext('Users')
    }
  }
]

const navItems = []

export default {
  appInfo,
  routes,
  translations,
  navItems
}
