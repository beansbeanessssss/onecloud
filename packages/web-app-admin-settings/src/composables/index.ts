export * from './actions'
export * from './usePagination'
