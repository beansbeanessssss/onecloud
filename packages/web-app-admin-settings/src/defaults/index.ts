export const supportedLogoMimeTypes = ['image/gif', 'image/jpeg', 'image/png']
export const perPageDefault = '50'
export const perPageStoragePrefix = 'admin-settings'
export const paginationOptions = ['20', '50', '100', '250']
