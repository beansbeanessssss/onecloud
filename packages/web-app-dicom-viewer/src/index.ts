import translations from '../l10n/translations.json'
import { AppWrapperRoute } from 'web-pkg/src/components/AppTemplates/AppWrapperRoute'
import DicomViewer from './DicomViewer.vue'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const routes = [
  {
    path: '/:driveAliasAndItem(.*)?',
    component: AppWrapperRoute(DicomViewer, {
      applicationId: 'dicom-viewer',
      urlForResourceOptions: {
        //disposition: 'inline'
      }
    }),
    name: 'dicom-viewer',
    meta: {
      authContext: 'hybrid',
      title: $gettext('DICOM Viewer'),
      patchCleanPath: true
    }
  }
]

const appInfo = {
  name: $gettext('DICOM Viewer'),
  id: 'dicom-viewer',
  icon: 'resource-type-medical',
  iconFillType: 'fill',
  iconColor: 'var(--oc-color-icon-medical)',
  extensions: [
    {
      extension: 'dcm',
      routeName: 'dicom-viewer'
    }
  ]
}

export default {
  appInfo,
  routes,
  translations
}
