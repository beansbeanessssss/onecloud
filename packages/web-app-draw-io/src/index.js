import translations from '../l10n/translations'
import App from './App.vue'

const routes = [
  {
    name: 'edit',
    path: '/edit/:filePath',
    components: {
      fullscreen: App
    },
    meta: { hideHeadbar: true }
  }
]

const appInfo = {
  name: 'Draw.io',
  id: 'draw-io',
  icon: 'grid_on',
  extensions: [
    {
      extension: 'drawio',
      newTab: true,
      routeName: 'draw-io-edit',
      newFileMenu: {
        menuTitle($gettext) {
          return $gettext('New draw.io documentâ€¦')
        }
      },
      routes: [
        'files-spaces-storage',
        'files-common-favorites',
        'files-shares-with-others',
        'files-shares-with-me',
        'files-public-files'
      ]
    },
    {
      extension: 'vsdx',
      newTab: true,
      routeName: 'draw-io-edit',
      routes: [
        'files-spaces-storage',
        'files-common-favorites',
        'files-shares-with-others',
        'files-shares-with-me',
        'files-public-files'
      ]
    }
  ]
}

export default {
  appInfo,
  routes,
  translations
}
