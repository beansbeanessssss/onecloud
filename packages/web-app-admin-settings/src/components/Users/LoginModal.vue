<template>
  <div>
    <oc-select
      :model-value="selectedOption"
      :label="$gettext('Login')"
      :options="options"
      :placeholder="$gettext('Select...')"
      :warning-message="
        currentUserSelected ? $gettext('Your own login status will remain unchanged.') : ''
      "
      @update:model-value="changeSelectedOption"
    />
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, onMounted, PropType, ref, unref, watch } from 'vue'
import { useGettext } from 'vue3-gettext'
import { User } from '@ownclouders/web-client/src/generated'
import { useClientService, useStore, useEventBus, Modal } from '@ownclouders/web-pkg'

type LoginOption = {
  label: string
  value: boolean
}

export default defineComponent({
  name: 'LoginModal',
  props: {
    modal: { type: Object as PropType<Modal>, required: true },
    users: {
      type: Array as PropType<User[]>,
      required: true
    }
  },
  emits: ['update:confirmDisabled'],
  setup(props, { emit, expose }) {
    const store = useStore()
    const clientService = useClientService()
    const eventBus = useEventBus()
    const { $gettext, $ngettext } = useGettext()

    const selectedOption = ref<LoginOption>()
    const options = ref([
      { label: $gettext('Allowed'), value: true },
      { label: $gettext('Forbidden'), value: false }
    ])

    watch(
      selectedOption,
      () => {
        emit('update:confirmDisabled', !unref(selectedOption))
      },
      { immediate: true }
    )

    const changeSelectedOption = (option: LoginOption) => {
      selectedOption.value = option
    }

    const currentUserSelected = computed(() => {
      return props.users.some((u) => u.id === store.getters.user.uuid)
    })

    onMounted(() => {
      if (props.users.every((u) => u.accountEnabled !== false)) {
        selectedOption.value = unref(options).find(({ value }) => !!value)
      } else if (props.users.every((u) => u.accountEnabled === false)) {
        selectedOption.value = unref(options).find(({ value }) => !value)
      }
    })

    const onConfirm = async () => {
      const affectedUsers = props.users.filter(({ id }) => store.getters.user.uuid !== id)
      const client = clientService.graphAuthenticated
      const promises = affectedUsers.map(({ id }) =>
        client.users.editUser(id, { accountEnabled: unref(selectedOption).value })
      )
      const results = await Promise.allSettled(promises)

      const succeeded = results.filter((r) => r.status === 'fulfilled') as any
      if (succeeded.length) {
        const title =
          succeeded.length === 1 && affectedUsers.length === 1
            ? $gettext('Login for user "%{user}" was edited successfully', {
                user: affectedUsers[0].displayName
              })
            : $ngettext(
                '%{userCount} user login was edited successfully',
                '%{userCount} users logins edited successfully',
                succeeded.length,
                { userCount: succeeded.length.toString() },
                true
              )
        store.dispatch('showMessage', { title })
      }

      const failed = results.filter((r) => r.status === 'rejected')
      if (failed.length) {
        failed.forEach(console.error)

        const title =
          failed.length === 1 && affectedUsers.length === 1
            ? $gettext('Failed edit login for user "%{user}"', {
                user: affectedUsers[0].displayName
              })
            : $ngettext(
                'Failed to edit %{userCount} user login',
                'Failed to edit %{userCount} user logins',
                failed.length,
                { userCount: failed.length.toString() },
                true
              )
        store.dispatch('showErrorMessage', {
          title,
          errors: (failed as PromiseRejectedResult[]).map((f) => f.reason)
        })
      }

      try {
        const usersResponse = await Promise.all(
          succeeded.map(({ value }) => client.users.getUser(value.data.id))
        )

        eventBus.publish(
          'app.admin-settings.users.update',
          usersResponse.map(({ data }) => data)
        )
      } catch (e) {
        console.error(e)
      }
    }

    expose({ onConfirm })

    return {
      selectedOption,
      options,
      changeSelectedOption,
      currentUserSelected,

      // unit tests
      onConfirm
    }
  }
})
</script>
