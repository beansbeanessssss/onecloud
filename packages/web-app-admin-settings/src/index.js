import translations from '../l10n/translations'
import Users from './views/Users.vue'
import Groups from './views/Groups.vue'
import Spaces from './views/Spaces.vue'
// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Administration Settings'),
  id: 'admin-settings',
  icon: 'settings-4',
  isFileEditor: false
}

const routes = [
  {
    path: '/',
    redirect: () => {
      const permissionManager = window.Vue.$permissionManager
      if (permissionManager.hasUserManagement()) {
        return { name: 'admin-settings-users' }
      }
      return { name: 'admin-settings-spaces' }
    }
  },
  {
    path: '/users',
    name: 'admin-settings-users',
    component: Users,
    meta: {
      authContext: 'user',
      title: $gettext('Users')
    }
  },
  {
    path: '/groups',
    name: 'admin-settings-groups',
    component: Groups,
    meta: {
      authContext: 'user',
      title: $gettext('Groups')
    }
  },
  {
    path: '/spaces',
    name: 'admin-settings-spaces',
    component: Spaces,
    meta: {
      authContext: 'user',
      title: $gettext('Spaces')
    }
  }
]

const navItems = [
  {
    name: $gettext('Users'),
    icon: 'user',
    route: {
      path: `/${appInfo.id}/users?`
    },
    enabled: () => {
      const permissionManager = window.Vue.$permissionManager
      return permissionManager.hasUserManagement()
    }
  },
  {
    name: $gettext('Groups'),
    icon: 'group-2',
    route: {
      path: `/${appInfo.id}/groups?`
    },
    enabled: () => {
      const permissionManager = window.Vue.$permissionManager
      return permissionManager.hasUserManagement()
    }
  },
  {
    name: $gettext('Spaces'),
    icon: 'layout-grid',
    route: {
      path: `/${appInfo.id}/spaces?`
    },
    enabled: () => {
      const permissionManager = window.Vue.$permissionManager
      return permissionManager.hasSpaceManagement()
    }
  }
]

export default {
  appInfo,
  routes,
  translations,
  navItems
}
