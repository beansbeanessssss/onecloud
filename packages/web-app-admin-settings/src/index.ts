import translations from '../l10n/translations.json'
import General from './views/General.vue'
import Users from './views/Users.vue'
import Groups from './views/Groups.vue'
import Spaces from './views/Spaces.vue'
// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Administration Settings'),
  id: 'admin-settings',
  icon: 'settings-4',
  isFileEditor: false
}

// FIXME: a better way to access this is needed
const permissionManager = () => (window as any).__$permissionManager

const routes = [
  {
    path: '/',
    redirect: () => {
      if (permissionManager().hasSystemManagement()) {
        return { name: 'admin-settings-general' }
      }
      return { name: 'admin-settings-spaces' }
    }
  },
  {
    path: '/general',
    name: 'admin-settings-general',
    component: General,
    meta: {
      authContext: 'user',
      title: $gettext('General')
    }
  },
  {
    path: '/users',
    name: 'admin-settings-users',
    component: Users,
    meta: {
      authContext: 'user',
      title: $gettext('Users')
    }
  },
  {
    path: '/groups',
    name: 'admin-settings-groups',
    component: Groups,
    meta: {
      authContext: 'user',
      title: $gettext('Groups')
    }
  },
  {
    path: '/spaces',
    name: 'admin-settings-spaces',
    component: Spaces,
    meta: {
      authContext: 'user',
      title: $gettext('Spaces')
    }
  }
]

const navItems = [
  {
    name: $gettext('General'),
    icon: 'settings-4',
    route: {
      path: `/${appInfo.id}/general?`
    },
    enabled: () => {
      return permissionManager().hasSystemManagement()
    }
  },
  {
    name: $gettext('Users'),
    icon: 'user',
    route: {
      path: `/${appInfo.id}/users?`
    },
    enabled: () => {
      return permissionManager().hasUserManagement()
    }
  },
  {
    name: $gettext('Groups'),
    icon: 'group-2',
    route: {
      path: `/${appInfo.id}/groups?`
    },
    enabled: () => {
      return permissionManager().hasUserManagement()
    }
  },
  {
    name: $gettext('Spaces'),
    icon: 'layout-grid',
    route: {
      path: `/${appInfo.id}/spaces?`
    },
    enabled: () => {
      return permissionManager().hasSpaceManagement()
    }
  }
]

export default {
  appInfo,
  routes,
  translations,
  navItems
}
