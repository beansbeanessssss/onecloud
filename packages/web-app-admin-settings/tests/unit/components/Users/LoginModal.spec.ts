import LoginModal from '../../../../src/components/Users/LoginModal.vue'
import {
  createStore,
  defaultComponentMocks,
  defaultPlugins,
  defaultStoreMockOptions,
  mockAxiosResolve,
  shallowMount
} from 'web-test-helpers'
import { mock } from 'jest-mock-extended'
import { User } from '@ownclouders/web-client/src/generated'
import { Modal, eventBus } from '@ownclouders/web-pkg'

describe('LoginModal', () => {
  it('renders the input including two options', () => {
    const { wrapper } = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('shows a warning when the current user is being selected', () => {
    const { wrapper } = getWrapper([mock<User>({ id: '1' })])
    expect(wrapper.findComponent<any>('oc-select-stub').props('warningMessage')).toBeDefined()
  })
  describe('method "onConfirm"', () => {
    it('updates the login for all given users', async () => {
      const users = [mock<User>(), mock<User>()]
      const { wrapper, mocks, storeOptions } = getWrapper(users)
      mocks.$clientService.graphAuthenticated.users.editUser.mockResolvedValue(
        mockAxiosResolve({ id: 'e3515ffb-d264-4dfc-8506-6c239f6673b5' })
      )
      mocks.$clientService.graphAuthenticated.users.getUser.mockResolvedValue(
        mockAxiosResolve({ id: 'e3515ffb-d264-4dfc-8506-6c239f6673b5' })
      )

      const eventSpy = jest.spyOn(eventBus, 'publish')

      await wrapper.vm.onConfirm()
      expect(storeOptions.actions.showMessage).toHaveBeenCalled()
      expect(eventSpy).toHaveBeenCalled()
      expect(mocks.$clientService.graphAuthenticated.users.editUser).toHaveBeenCalledTimes(
        users.length
      )
    })
    it('omits the currently logged in user', async () => {
      const users = [mock<User>({ id: '1' }), mock<User>()]
      const { wrapper, mocks } = getWrapper(users)
      mocks.$clientService.graphAuthenticated.users.editUser.mockResolvedValue(
        mockAxiosResolve({ id: 'e3515ffb-d264-4dfc-8506-6c239f6673b5' })
      )
      mocks.$clientService.graphAuthenticated.users.getUser.mockResolvedValue(
        mockAxiosResolve({ id: 'e3515ffb-d264-4dfc-8506-6c239f6673b5' })
      )

      await wrapper.vm.onConfirm()
      expect(mocks.$clientService.graphAuthenticated.users.editUser).toHaveBeenCalledTimes(1)
    })
    it('should show message on error', async () => {
      jest.spyOn(console, 'error').mockImplementation(() => undefined)

      const users = [mock<User>(), mock<User>()]
      const { wrapper, mocks, storeOptions } = getWrapper(users)
      mocks.$clientService.graphAuthenticated.users.editUser.mockRejectedValue(new Error(''))
      mocks.$clientService.graphAuthenticated.users.getUser.mockRejectedValue(new Error(''))

      await wrapper.vm.onConfirm()
      expect(storeOptions.actions.showErrorMessage).toHaveBeenCalled()
    })
  })
})

function getWrapper(users = [mock<User>()]) {
  const mocks = defaultComponentMocks()
  const storeOptions = defaultStoreMockOptions
  const store = createStore(storeOptions)

  return {
    mocks,
    storeOptions,
    wrapper: shallowMount(LoginModal, {
      props: {
        modal: mock<Modal>(),
        users
      },
      global: {
        provide: mocks,
        plugins: [...defaultPlugins(), store]
      }
    })
  }
}
