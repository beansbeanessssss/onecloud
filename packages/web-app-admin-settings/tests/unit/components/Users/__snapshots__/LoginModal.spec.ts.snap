// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal renders the input including two options 1`] = `
<div>
  <oc-select-stub clearable="false" disabled="false" filter="[Function]" fixmessageline="false" getoptionlabel="[Function]" id="oc-select-1" label="Login" loading="false" multiple="false" optionlabel="label" options="[object Object],[object Object]" placeholder="Select..." readonly="false" searchable="true" warningmessage=""></oc-select-stub>
</div>
`;
